internal/controller/gardener_cluster_controller.go

```diff
@@ -0,0 +1,179 @@
+//+kubebuilder:rbac:groups=infrastructuremanager.kyma-project.io,resources=clusters/status,verbs=get;update;patch
+//+kubebuilder:rbac:groups=infrastructuremanager.kyma-project.io,resources=clusters/finalizers,verbs=update
+
+// Reconcile is part of the main kubernetes reconciliation loop which aims to
+// move the current state of the cluster closer to the desired state.
+// the GardenerCluster object against the actual cluster state, and then
+// perform operations to make the cluster state reflect the state specified by
+// the user.
+//
+// For more details, check Reconcile and its Result here:
+// - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.15.0/pkg/reconcile
+func (r *GardenerClusterController) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) { //nolint:revive
+	r.log.Info("Starting reconciliation loop")
+
+	var cluster infrastructuremanagerv1.GardenerCluster
+
+	err := r.Client.Get(ctx, req.NamespacedName, &cluster)
+
+	if err != nil {
+		r.log.Error(err, "could not get the CR for "+req.NamespacedName.Name)
+		return r.ResultWithoutRequeue(), err
+	}
+
+	secret, err := r.getSecret(cluster.Spec.Shoot.Name)
+	if err != nil {
+		r.log.Error(err, "could not get the Secret for "+cluster.Spec.Shoot.Name)
+
+		if !k8serrors.IsNotFound(err) {
+			return r.ResultWithoutRequeue(), err
+		}
+	}
+
+	if secret == nil {
+		r.log.Error(err, "Secret not found, and will be created")
+
+		err = r.createSecret(ctx, cluster)
+
+		if err != nil {
+			return r.ResultWithoutRequeue(), err
+		}
+	}
+
+	return ctrl.Result{}, nil
+}
+
+func (r *GardenerClusterController) ResultWithoutRequeue() ctrl.Result {
+	return ctrl.Result{
+		Requeue: false,
+	}
+}
+
+func (r *GardenerClusterController) getSecret(shootName string) (*corev1.Secret, error) {
+	var secretList corev1.SecretList
+
+	shootNameSelector := client.MatchingLabels(map[string]string{
+		"kyma-project.io/shoot-name": shootName,
+	})
+
+	err := r.Client.List(context.Background(), &secretList, shootNameSelector)
+	if err != nil {
+		return nil, err
+	}
+
+	size := len(secretList.Items)
+
+	if size == 0 {
+		return nil, k8serrors.NewNotFound(schema.GroupResource{}, "") //nolint:all TODO: update with valid GroupResource
+	}
+
+	if size > 1 {
+		return nil, fmt.Errorf("unexpected numer of secrets found for shoot `%s`", shootName)
+	}
+
+	return &secretList.Items[0], nil
+}
+
+func (r *GardenerClusterController) createSecret(ctx context.Context, cluster infrastructuremanagerv1.GardenerCluster) error {
+	secret, err := r.newSecret(cluster)
+	if err != nil {
+		return err
+	}
+
+	return r.Client.Create(ctx, &secret)
+}
+
+func (r *GardenerClusterController) newSecret(cluster infrastructuremanagerv1.GardenerCluster) (corev1.Secret, error) {
+	labels := map[string]string{}
+
+	for key, val := range cluster.Labels {
+		labels[key] = val
+	}
+	labels["operator.kyma-project.io/managed-by"] = "lifecycle-manager"
```

Comment:
I think think it should be "infrastructure-manager" instead of "lifecycle-manager"